UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs

```diff
@@ -0,0 +1,91 @@
+﻿using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using UKHO.ExchangeSetService.Common.Configuration;
+using UKHO.ExchangeSetService.Common.Logging;
+using UKHO.ExchangeSetService.Common.Models.FileShareService.Request;
+using UKHO.ExchangeSetService.Common.Models.FileShareService.Response;
+
+namespace UKHO.ExchangeSetService.Common.Helpers
+{
+    public class FileShareService : IFileShareService
+    {
+        private readonly IFileShareServiceClient fileShareServiceClient;
+        private readonly IAuthTokenProvider authTokenProvider;
+        private readonly IOptions<FileShareServiceConfiguration> fileShareServiceConfig;
+        private readonly ILogger<FileShareService> logger;
+        private string batchExpiryDateTime;
```

Comment:
Do we really need `batchExpiryDateTime` as a field? Please can you check.

Reply:
Yes this is required as we need value of batchExpiryDateTime to be same when we create batch request and when we create batch response to be set for ESS response. Directly using DateTime.UtNow will change its value during execution time. Hence we have assigned its value to variable batchExpiryDateTime

ExpiryDate = DateTime.UtcNow.AddDays(1).ToString("yyyy-MM-ddTHH:mm:ssZ", CultureInfo.InvariantCulture),
batchExpiryDateTime =createBatchRequest.ExpiryDate;

createBatchResponse.ResponseBody.BatchExpiryDateTime = batchExpiryDateTime;

Reply:
@janvi13603 you can hold it in a local variable and pass it to `CreateBatchResponse` as a parameter.

Reply:
Done.